kernel/src/hil/uart.rs

```diff
@@ -117,7 +147,7 @@ pub trait Transmit<'a> {
     /// Transmit a single word of data asynchronously. The word length is
     /// determined by the UART configuration: it can be 6, 7, 8, or 9 bits long.
     /// If the `Result<(), ErrorCode>` is Ok(()), on completion,
-    /// `transmitted_word` will be called on the `TransmitClient`.
+    /// `transmitted_character` will be called on the `TransmitClient`.
     /// Other valid `Result<(), ErrorCode>` values are:
     ///  - OFF: The underlying hardware is not available, perhaps because
```

Comment:
Why this change of naming? I associate `character` strictly with one ASCII character, so 8 bits. OTOH, `word` has semantics of being strictly dependent on the hardware, which is exactly the case here.

Reply:
WDYT about making this strongly typed in a way that `Word` becomes an enum:
```rust
enum Word {
  Len6(u16),
  Len7(u16),
  Len8(u16),
  Len9(u16),
}
```
so that it becomes clearer for the HIL users that only a couple of lower bits of `word` argument is actually passed to the hardware, even though the type is much bigger (`u32` in the current code, `u16` as I propose, for the whole tagged enum to fit in a single `u32`).

Reply:
I think those are both good suggestions, but I think those should go into the UART TRD itself which specifies this interface. Feel free to create a PR for them!